libraries/Microsoft.Bot.Builder.Dialogs/DialogContextExtension.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="DialogContext"/>.
+    /// </summary>
+    public static class DialogContextExtension
+    {
+        private const string Turn = "turn";
+        private const string LocalePath = "$.activity.locale";
+
+        /// <summary>
+        /// Obtain the CultureInfo in DialogContext.
+        /// </summary>
+        /// <param name="dialogContext">The dialogContext to extract information.</param>
+        /// <returns>A <see cref="CultureInfo"/> representing the current locale.</returns>
+        public static CultureInfo EvalLocaleInfo(this DialogContext dialogContext)
```

Comment:
Maybe, call this GetLocale?

Reply:
And the return type should be string for better portability.

Reply:
changed